prov/sockets/include/sock.h

```diff
@@ -1211,6 +1218,14 @@ int sock_epoll_wait(struct sock_epoll_set *set, int timeout);
 int sock_epoll_get_fd_at_index(struct sock_epoll_set *set, int index);
 void sock_epoll_close(struct sock_epoll_set *set);
 
+int sock_epoll_map_init(struct sock_epoll_map *emap);
+int sock_epoll_map_destroy(struct sock_epoll_map *emap);
+int sock_epoll_map_add(struct sock_epoll_map *emap, int fd, void *ptr);
+int sock_epoll_map_del(struct sock_epoll_map *emap, int fd);
+void *sock_epoll_map_lookup(struct sock_epoll_map *emap, int fd);
+int sock_epoll_map_signal(struct sock_epoll_map *emap);
+int sock_epoll_map_wait(struct sock_epoll_map *emap);
```

Comment:
The basic idea behind this patch is to provide an interface that associates a pointer to a file descriptor returned by `epoll_wait`. This patch is a requirement for my big patch for the new connection manager and makes the code easier.

With epoll(7), you already can associate a pointer to a file descriptor. E.g:
`epoll_data_t evt;`
`evt.ptr = myHandle;`
`epoll_ctl(epoll_fd, EPOLL_CTL_ADD, fd, &evt) `
Then in `epoll_wait()` you can directly have access to `myHandle`.

While this feature is already implemented by `epoll`, Libfabric must support platforms where `epoll(7)` is not available but only `poll(2)` is. The problem is that `poll(2)` only returns a set of file descriptors, which is not convenient.

I've checked the file fi_util.h and I didn't find anything with similar functionalities.
If you prefer I can directly add my changes into the sock_epoll interface. Also, I think that the whole file `sockets/src/sock_epoll.c` should be moved into the general utility code.